src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -1,248 +1,477 @@
-					))}
-				</List>
-			)}
-			{acceptedFiles.map((file) => (
-				<List sx={{ listStyleType: 'disc' }}>
-					<ListItem
-						sx={{
-							py: 0,
-						}}
-					>
-						<ListItemText sx={{ display: 'list-item' }}>
-							<Typography component="span" fontWeight="bold">
-								{file.name}
-							</Typography>{' '}
-							({file.type}, {file.size} bytes)
-						</ListItemText>
-					</ListItem>
-				</List>
-			))}
-		</>
+								Maximum size: {fileSizeToReadable(maxSize)}
+							</Typography>
+						)}
+					</Stack>
+
+					{hideUploadedFiles && !!errorFiles.length && (
+						<Stack color="error.main" gap={1}>
+							{errorFiles
+								.flatMap((errorFile) => errorFile.errors)
+								.map((error) =>
+									getErrorMessage(
+										error.code,
+										error.message,
+										maxSize ?? Infinity,
+									),
+								)}
+						</Stack>
+					)}
+					{!hideUploadedFiles && (
+						<Stack gap={1}>
+							{files.map((file, index) => (
+								<Stack
+									direction="row"
+									justifyContent="space-between"
+									alignItems="center"
+									gap={2}
+									bgcolor={designTokens.color.bg.value}
+									p={3}
+									borderRadius="8px"
+									{...(mobile && { flexWrap: 'wrap' })}
+								>
+									<Stack
+										direction="row"
+										{...(mobile && { justifyContent: 'space-between' })}
+										gap={2}
+										alignItems="center"
+										width="100%"
+									>
+										<Stack
+											direction="row"
+											gap={2}
+											alignItems="center"
+											maxWidth="45%"
+										>
+											<FontAwesomeIcon icon={getFileTypeIcon(file.type)} />
+											<Typography
+												{...(mobile
+													? { maxWidth: '100%' }
+													: { width: '150px' })}
+												noWrap
+											>
+												{file.name}
+											</Typography>
+										</Stack>
+										{file.loadingPercentage == null && (
+											<LinearProgress
+												variant="determinate"
+												value={file.loadingPercentage ?? 50}
+												sx={{
+													width: mobile ? '100%' : '150px',
+													...(mobile && { maxWidth: '45%' }),
+													height: '8px',
+												}}
+											/>
+										)}
+									</Stack>
+									<Stack
+										direction="row"
+										{...(mobile
+											? { justifyContent: 'space-between', width: '100%' }
+											: { gap: 2 })}
+										alignItems="center"
+									>
+										<Stack direction="row" gap={2}>
+											<Typography color={designTokens.color.text.subtle.value}>
+												{file.type}
+											</Typography>
+											<Typography color={designTokens.color.text.subtle.value}>
+												{fileSizeToReadable(file.size)}
+											</Typography>
+										</Stack>
+										<IconButtonWithTracking
+											eventName="FileWidget: Delete uploaded file"
+											aria-label="Delete uploaded file"
+											onClick={() => removeFile(index)}
+											color="secondary"
```

Comment:
@drskullster please confirm we should indeed be changing the color of this button to "secondary"

Reply:
it should be the default color @myarmolinsky, but I realize it's not the same in my mockup than in the design system. I say just remove the `color` prop and roll with that, we'll adapt in a future PR if needs be.